src/models/customs-declarations.js

```diff
@@ -99,34 +101,30 @@ export const getCustomsDeclarationOpenState = (finalisation) => !(
   (finalisation.finalState === '1' || finalisation.finalState === '2')
 )
 
+const createLegacyDecision = (commodity, check, document, decisionCheck) => ({
+  itemNumber: commodity.itemNumber,
+  documentReference: document?.documentReference || null,
+  checkCode: check.checkCode,
+  decisionCode: decisionCheck?.decisionCode,
+  decisionReason: decisionCheck?.decisionReasons?.length > 0 ? decisionCheck?.decisionReasons[0] : null,
```

Comment:
Hard to wrap your head round :) Could we try something like:
```suggestion
  decisionReason: decisionCheck?.decisionReasons?.[0] ?? null,
```